pkg/healthz/healthz.go

```diff
@@ -0,0 +1,78 @@
+// Package healthz implements healthz handlers.
+package healthz
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	pb "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1"
+	"sigs.k8s.io/aws-encryption-provider/pkg/plugin"
+)
+
+// NewHandler returns a new healthz handler.
+func NewHandler(srv *plugin.Plugin,
+	cli pb.KeyManagementServiceClient,
+	checkPeriod time.Duration,
+) http.Handler {
+	return &handler{
+		srv:         srv,
+		cli:         cli,
+		checkPeriod: checkPeriod,
+	}
+}
+
+type handler struct {
+	srv         *plugin.Plugin
+	cli         pb.KeyManagementServiceClient
+	checkPeriod time.Duration
+
+	lastMu   sync.Mutex
+	lastErr  error
+	lastTime time.Time
+}
+
+func (hd *handler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	res, err := plugin.Check(hd.cli)
```

Comment:
What is the value of adding a version check? This is just static info right now (https://github.com/kubernetes-sigs/aws-encryption-provider/blob/master/pkg/plugin/plugin.go#L65)

Reply:
The plugin check was there before this PR. We can remove if we replace it with kms API check.